trunk/src/app/srs_app_gb28181.cpp

```diff
@@ -306,35 +306,7 @@ srs_error_t SrsGb28181PsRtpProcessor::on_rtp_packet(const sockaddr* from, const
         if(key != cache_ps_rtp_packet.end())
         {
             SrsGb28181RtmpMuxer* muxer = NULL;
-            //First, search according to the channel_id. Otherwise, search according to the SSRC.
-            //Some channel_id are created by RTP pool, which are different ports.
-            //No channel_id are created by multiplexing ports, which are the same port
-            if (!channel_id.empty()){
-                muxer = _srs_gb28181->fetch_rtmpmuxer(channel_id);
-            }else {
-                muxer = _srs_gb28181->fetch_rtmpmuxer_by_ssrc(pkt.ssrc);
-            }
-
-            //auto crate channel
-            if (!muxer && config->auto_create_channel){
-                //auto create channel generated id
-                std::stringstream ss, ss1;
-                ss << "chid" << pkt.ssrc;
-                std::string tmp_id = ss.str();
-
-                SrsGb28181StreamChannel channel;
-                channel.set_channel_id(tmp_id);
-                channel.set_port_mode(RTP_PORT_MODE_FIXED);
-                channel.set_ssrc(pkt.ssrc);
-              
-                srs_error_t err2 = srs_success;
-                if ((err2 = _srs_gb28181->create_stream_channel(&channel)) != srs_success){
-                    srs_warn("gb28181: RtpProcessor create stream channel error %s", srs_error_desc(err2).c_str());
-                    srs_error_reset(err2);
-                };
-
-                muxer = _srs_gb28181->fetch_rtmpmuxer(tmp_id);
-            }
+            muxer = fetch_rtmpmuxer(channel_id,pkt.ssrc);       
```

Comment:
这是你调整后的代码吗？@xbpeng121
这儿改动很大呀？auto_create_channel被你弄掉了？

Reply:
这部分代码 
           `//First, search according to the channel_id. Otherwise, search according to the SSRC.
             //Some channel_id are created by RTP pool, which are different ports.
             //No channel_id are created by multiplexing ports, which are the same port
             if (!channel_id.empty()){
                 muxer = _srs_gb28181->fetch_rtmpmuxer(channel_id);
             }else {
                 muxer = _srs_gb28181->fetch_rtmpmuxer_by_ssrc(pkt.ssrc);
             }

             //auto crate channel
             if (!muxer && config->auto_create_channel){
                 //auto create channel generated id
                 std::stringstream ss, ss1;
                 ss << "chid" << pkt.ssrc;
                 std::string tmp_id = ss.str();

                 SrsGb28181StreamChannel channel;
                 channel.set_channel_id(tmp_id);
                 channel.set_port_mode(RTP_PORT_MODE_FIXED);
                 channel.set_ssrc(pkt.ssrc);

                 srs_error_t err2 = srs_success;
                 if ((err2 = _srs_gb28181->create_stream_channel(&channel)) != srs_success){
                     srs_warn("gb28181: RtpProcessor create stream channel error %s", srs_error_desc(err2).c_str());
                     srs_error_reset(err2);
                 };

                 muxer = _srs_gb28181->fetch_rtmpmuxer(tmp_id);
             }`
与fetch_rtmpmuxer(channel_id,pkt.ssrc)函数里面的代码基本完全一样。
只是在on_rtp_packet()函数里直接写，在on_rtp_packet_jitter()函数里面就将它们封装在了fetch_rtmpmuxer()函数里面。这里调用了fetch_rtmpmuxer()函数，并没有修改auto_create_channel的逻辑。@PieerePi